### ISSUE TITLE
Issues with locale 

### ISSUE THREAD

**Issue by achekulaev** [docksal/docksal#692]
Running `fin drush @stage.ipo status --format=list bootstrap` is returning this result:

```
bash: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
Successful
```

If we create a Dockerfile:

```Dockerfile
# Use a stock Docksal image as the base
FROM docksal/cli:php7.1

RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
    locale-gen && \
    update-locale LANG=en_US.UTF-8

ENV LC_ALL en_US.UTF-8

USER root
```

Then we no longer have the issue. The Docksal cli image specifically sets it to C.UTF-8. I'm not sure why. We can do this Dockerfile way, but I would prefer if we didn't have to do that.

**Comment by lmakarov** [comment]
I'm not able to reproduce this using the docksal/drupal8 example repo.

```
docker@cli:/var/www/docroot$ drush st bootstrap --format=list
Successful
```

Tried with the latest stable as well as edge version of cli

```
CLI_IMAGE=docksal/cli:2.2-php7.1
CLI_IMAGE=docksal/cli:edge-php7.1
```

**Comment by lmakarov** [comment]
Looks like the [code](https://github.com/docksal/service-cli/blob/6f9043504b7439d521e263cf6c10e378fe3995f6/7.1/Dockerfile#L34-L39) we have to set locales is not even necessary, since the parent images (e.g. `php:7.1-fpm`) already come with the following locales installed:

```
$ docker run --rm -it php:7.1-fpm locale -a
C
C.UTF-8
POSIX
```

In the Debian image (parent of php:7.1-fpm) README they mention about locales [here](https://github.com/docker-library/docs/tree/master/debian#locales), saying `C.UTF-8` should work in most cases.

We can probably add `en_US.UTF-8` and make it the default, but I'd like to better understand what's trying to override the locale and causing the error in your case, @achekulaev

**Comment by achekulaev** [comment]
@lmakarov it's not my case. It's @shelane 's

**Comment by shelane** [comment]
Let me know what information I can give you. I’m not sure why we getting this error just trying to do a drush status. It wasn’t when trying to do drush status to a site in the container, but from the container connecting to our server. We call the status in our site-import command to verify the site exists on the server.

**Comment by achekulaev** [comment]
Looks like [this thread](https://github.com/tianon/docker-brew-debian/issues/45) has a lot of relevant info.
I guess the server Shelane connects to has different set of locales, wider than the one we have in cli.

**Comment by lmakarov** [comment]
@shelane please run the following in `fin bash`:

```
sudo localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
export LANG en_US.utf8
```

Then, in the same `fin bash` session, test again with your drush command:

```
drush @stage.ipo status --format=list bootstrap
```

**Comment by shelane** [comment]
The response to the second command was:
`bash: export: `en_US.utf8': not a valid identifier`
So I am still getting:
```
docker@cli:/var/www$ drush @stage.ipo status --format=list bootstrap
bash: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
Successful
```

**Comment by lmakarov** [comment]
@shelane Apologies, the was a typo. Try `export LANG=en_US.utf8`

**Comment by shelane** [comment]
OK. I ran it with the `export LANG=en_US.utf8` and I still get the same message as before when running the drush command.

**Comment by lmakarov** [comment]
Try `export LC_ALL=en_US.UTF-8` then the drush command.

**Comment by shelane** [comment]
That did it. The drush command now only returns "Successful"

**Comment by frederickjh** [comment]
I had this issue (**bash: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)**" a long time ago doing an ssh deploy from Docksal to a remote host via ssh. 
Just ran into this error again. I looked at the older Docksal environment where I had previously fixed it and found that I had added the following to `docksal-local.yml`.

```yaml
services:
  cli:
    environment:
      - LANG=en_US.UTF-8
      - LC_ALL=
      - LANGUAGE=en_US.UTF-8
      - LC_CTYPE=en_US.UTF-8
    volumes:
```
I drop this in here as when I searched this was the issue I found and maybe this information will be helpful to someone else, in that you do not need to modify the `Dockerfile` but can add the needed environment variables in the `docksal-local.yml` file.
